server/applicationPattern/applicationPattern/onfModel/models/layerProtocols/HttpClientInterface.js

```diff
@@ -145,6 +181,44 @@ class HttpClientInterface extends layerProtocol {
         return httpClientUuid != undefined;
     }
 
+    /**
+     * @description This function checks if application exists other than old release and new release
+     * @param {String} applicationName : name of the application.
+     * @param {String} releaseNumber : release number of the application.
+     * @returns {Promise<boolean>} true|false
+     **/
+    static async isApplicationExistsExcludingOldReleaseAndNewRelease(applicationName, releaseNumber) {
+        let httpClientUuid = await HttpClientInterface.getHttpClientUuidExcludingOldReleaseAndNewRelease(applicationName, releaseNumber);
+        return httpClientUuid != undefined;
+    }
+
+    /**
+     * @description This function returns the uuid of the http-client-interface for the application-name and release-number.If release number
+     * is not provided , then only the application name will be checked other than old release and new release
+     * @param {String} applicationName : name of the application.
+     * @param {String} releaseNumber : release number of the application.
+     * @returns {Promise<List>} undefined|httpClientUuid
+     **/
+        static async getHttpClientUuidExcludingOldReleaseAndNewRelease(applicationName, releaseNumber) {
+            return new Promise(async function (resolve, reject) {
+                try {
+                    let httpClientUuid;
+                    let httpClientUuidList = await HttpClientInterface.getHttpClientUuidListAsync(applicationName, releaseNumber);
+                    if(httpClientUuidList !== undefined){
+                        for(let i=0 ; i<httpClientUuidList.length ; i++){
+                              let uuid = httpClientUuidList[i];
+                              if(!(await LogicalTerminationPoint.ishttpClientUuidInstanceofOldReleaseOrNewRelease(uuid))){
+                                 httpClientUuid = uuid; 
+                              }
+                        }
+                    }
+                    resolve(httpClientUuid)
+                } catch (error) {
+                    console.log(error)
+                }
+            })
+        }
```

Comment:
Formatting here is all wrong. Also without the boilerplate code it can be written in about 10 lines.